File: records/count_lines.pl
use v5.26;
use warnings;
use List::Util qw(sum);

package Record {
    use Class::XSAccessor
      constructor => 'new',
      accessors   => ['name', 'count'];
}

sub create_records {
    my $count = scalar(grep { index(lc substr($_, 1, 5), 'bc') >= 0 } @_);
    return Record->new(
        name  => $_[0],
        count => $count
    );
}

sub main {
    my @records;
    while (my $line = <STDIN>) {
        push @records, create_records(split(/\t/, $line));
    }
    say sum(map { $_->count } @records);
}

main();

File: records/count_lines.kt
import java.io.File
import kotlin.text.*

data class Record(val name: String, val count: Int)
fun createRecord(vals: List<String>): Record {
    val c = vals.count {
        it.substring(1,4).contains("bc", true)
    }
    return Record(vals[0], c)
}

fun main() {
    var records: MutableList<Record> = mutableListOf<Record>()
    File("/dev/stdin").forEachLine(Charsets.US_ASCII) {
        records.add(createRecord(it.split('\t')))
    }
    println(records.sumBy{it.count})
}